### PR TITLE
US9986: Maestro Event Tracking

### PR DESCRIPTION
**Pull request by lukemoderwell**
**What this PR would do:** 
- create CRDS.DataTracker class
- track clicks on elements with data-track-click attributes
- track searches on forms with data-track-search attribute
- send this information to Astronomer.io so it can be routed to other analytics tools

### CODE REVIEW

File: apps/crossroads_interface/test/js/dataTracker_test.js [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      <div class="input-group">
+        <input class="form-control" placeholder="Search by address or zip code" type="text"><span class="input-group-btn"> <button class="btn btn-secondary" type="submit"></button></span>
+      </div>
+  </form>
+  `;
+
+  let dataTracker;
+  let btn1;
+  let btn2;
+  let search1;
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/test/js/dataTracker_test.js]: It doesn't look like search1 is getting used in any of the tests. I see it get set in the beforeEach() and unset in the afterEach(), but I don't see it getting used in either of the test cases.

Also, it looks like only btn1 is getting used in the tests. We should remove any variable declarations and assignments for variables that aren't getting used.

File: apps/crossroads_interface/web/static/js/dataTracker.js [hunk 2]
```diff
@@ -0,0 +1,53 @@
+/* eslist no-plusplus: 0 */
+/* global CRDS analytics */
+
+window.CRDS = window.CRDS || {};
+
+CRDS.DataTracker = class DataTracker {
+  constructor() {
+    this.clickTrackable = undefined;
+    this.searchTrackable = undefined;
+    this.analytics = analytics;
```
Comment (mjvcallibrity) [inline apps/crossroads_interface/web/static/js/dataTracker.js]: Where is the `analytics` coming from that is getting assigned to `this.analytics`?

Comment (lukemoderwell) [reply apps/crossroads_interface/web/static/js/dataTracker.js]: @mjvcallibrity It's in the global namespace

Comment (richardbarker) [reply apps/crossroads_interface/web/static/js/dataTracker.js]: I think it's getting pulled in through google tag manager, and is available in global scope.

Comment (mjvcallibrity) [reply apps/crossroads_interface/web/static/js/dataTracker.js]: Great. Thanks!

**Pull request merged**